adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceBannerAd.m

```diff
@@ -0,0 +1,129 @@
+//
+//  GADMAdapterIronSourceBannerAd.m
+//  ISMedAdapters
+//
+//  Created by alond on 11/12/2022.
+//  Copyright © 2022 ironSource Ltd. All rights reserved.
+//
+
+#import "GADMAdapterIronSourceBannerAd.h"
+#import "GADMAdapterIronSourceConstants.h"
+#import "GADMAdapterIronSourceBannerDelegate.h"
+#import "GADMAdapterIronSourceUtils.h"
+#import "GADMediationAdapterIronSource.h"
+#import "ISMediationManager.h"
+
+@interface GADMAdapterIronSourceBannerAd () <GADMediationBannerAd,GADMAdapterIronSourceBannerDelegate,ISDemandOnlyBannerDelegate>
+
+// The completion handler to call when the ad loading succeeds or fails.
+@property(copy, nonatomic) GADMediationBannerLoadCompletionHandler adLoadCompletionHandler;
+
+// Ad configuration for the ad to be rendered.
+@property(weak, nonatomic) GADMediationBannerAdConfiguration *adConfiguration;
+
+// An ad event delegate to invoke when ad rendering events occur.
+@property(weak, nonatomic) id<GADMediationBannerAdEventDelegate> adEventDelegate;
+
+/// Holds the ID of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceID;
+
+/// Holds the state of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceState;
+
+@property (nonatomic, strong) ISDemandOnlyBannerView *iSDemandOnlyBannerView;
+
+@end
+
+@implementation GADMAdapterIronSourceBannerAd
+
+- (instancetype)initWithGADMediationBannerConfiguration:(GADMediationBannerAdConfiguration *)adConfiguration
+                                      completionHandler:(GADMediationBannerLoadCompletionHandler)completionHandler {
+    self = [super init];
+    if (self) {
+        _adLoadCompletionHandler = completionHandler;
+        _adConfiguration = adConfiguration;
+        // Default instance ID
+        self.instanceID = GADMIronSourceDefaultInstanceId;
+        // Default instance state
+        self.instanceState = GADMAdapterIronSourceInstanceStateStart;
+    }
+    
+    return self;
+}
+
+- (void)renderBannerForAdConfig:(nonnull GADMediationBannerLoadCompletionHandler)handler {
```

Comment:
Per another comment, can combine these two methods together, see [this example](https://github.com/googleads/googleads-mobile-ios-mediation/blob/main/adapters/Mintegral/MintegralAdapter/Waterfall/GADMAdapterMintegralBannerAdLoader.m).